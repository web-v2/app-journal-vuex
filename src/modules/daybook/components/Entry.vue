<template>
    <div>
        <div class="entry-container mb-3 pointer p-2" @click="$router.push({name: 'entry', params: {id: '1s5a1s5a8s7arfg54s4a5'}})">            
            <!-- titulo-->
            <div class="entry-title d-flex">
                <span class="text-success fs-5 fw-bold">01</span>
                <span class="mx-1 fs-5">Agosto</span>
                <span class="mx-2 fw-light">2022, lunes</span>
            </div>
            <div class="entry-description">
                Lorem ipsum dolor, sit amet consectetur adipisicing elit. Neque blanditiis dicta excepturi at alias provident sequi labore iusto magnam dolorem, fugit saepe molestiae recusandae laboriosam eum. Nesciunt pariatur ex facilis.
            </div>
        </div>
    </div>
</template>
<script>
export default {

}
</script>
<style lang="scss" scoped>
.entry-container{
    border-bottom: 1px solid #2c3e50;
    transition: 0.2s all ease-in;
    &:hover{
        background-color: lighten($color: grey, $amount: 45);
        transition: 0.2s all ease-in;
    }
    .entry-description{
        font-size: 14px;
    }
}

</style>