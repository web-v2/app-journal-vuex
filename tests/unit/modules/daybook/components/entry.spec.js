import { shallowMount } from "@vue/test-utils";
import Entry from '@/modules/daybook/components/Entry';
import {journalState} from '../../../mock-data/test-journal-state'

describe('Pruebas en el Entry Component', () => {
    const mockRouter = {push: jest.fn()}
    const wrapper = shallowMount(Entry, {
        props: {
            entry: journalState.entries[0]
        }, 
        global: {
            mocks: {
                $router: mockRouter
            }
        }
    })

    test('debe hacer match con el snapshot', () => {       
        expect(wrapper.html() ).toMatchSnapshot()
    });

    test('debe redireccionar al hacer click en el entry-container', () => {
        const entryContainer = wrapper.find('.entry-container')
        entryContainer.trigger('click')
        expect(mockRouter.push).toHaveBeenCalledWith({
            name: 'entry', params: {id: journalState.entries[0].id }
        })
    });

    test('pruebas en las propiedades computadas', () => {
        expect( wrapper.vm.day ).toBe(17)
        expect( wrapper.vm.month ).toBe('Noviembre')
        expect( wrapper.vm.yearDate ).toBe('2022, Jueves')
    });
});