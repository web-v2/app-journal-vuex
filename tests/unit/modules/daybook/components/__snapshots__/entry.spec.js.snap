// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el Entry Component debe hacer match con el snapshot 1`] = `
<div>
  <div class="entry-container mb-3 pointer p-2">
    <!-- titulo-->
    <div class="entry-title d-flex"><span class="text-success fs-5 fw-bold">17</span><span class="mx-1 fs-5">Noviembre</span><span class="mx-2 fw-light">2022, Jueves</span></div>
    <div class="entry-description">Mediante el siguiente POST se realizara la carga de im√°genes desde el FRONT al BACK en el servicio de Cloudinary.</div>
  </div>
</div>
`;
